app/src/main/java/blockeq/com/stellarwallet/helpers/SupportedMnemonic.kt

```diff
@@ -0,0 +1,67 @@
+package blockeq.com.stellarwallet.helpers
+
+import blockeq.com.stellarwallet.BuildConfig
+import com.soneso.stellarmnemonics.WalletException
+import com.soneso.stellarmnemonics.derivation.Ed25519Derivation
+import com.soneso.stellarmnemonics.mnemonic.MnemonicException
+import com.soneso.stellarmnemonics.util.PrimitiveUtil
+import org.stellar.sdk.KeyPair
+import javax.crypto.SecretKeyFactory
+import javax.crypto.spec.PBEKeySpec
+
+class SupportedMnemonic {
+    companion object {
+        const val ALGORITHM_API_19 = "PBKDF2withHmacSHA1And8BIT"
+        const val ALGORITHM_API_26 = "PBKDF2WithHmacSHA512"
+
+        fun supportedSeed(mnemonic: CharArray, passphrase: CharArray?): ByteArray {
+            var saltChars = charArrayOf('m', 'n', 'e', 'm', 'o', 'n', 'i', 'c')
+            if (passphrase != null) {
+                saltChars = PrimitiveUtil.concatCharArrays(saltChars, passphrase)
+            }
+
+            val salt = PrimitiveUtil.toBytes(saltChars)
+
+            try {
+                val ks = PBEKeySpec(mnemonic, salt, 2048, 512)
+                val skf = SecretKeyFactory.getInstance(ALGORITHM_API_19)
+                return skf.generateSecret(ks).encoded
+            } catch (var6: Exception) {
+                throw MnemonicException("Fatal error when generating seed from mnemonic!")
+            }
+        }
+
+        @Throws(MnemonicException::class)
+        fun createSeed(mnemonic: CharArray, passphrase: CharArray?): ByteArray {
```

Comment:
This is repetitive code from above...we only need to change the Algorithm. We can pass in the API level as a parameter and switch the Algorithm. This way we only need one function to create seed that takes in API as a parameter.